framework/src/consumption/lib/quicksight/quicksight-subscription.ts

```diff
@@ -0,0 +1,232 @@
+  */
+  readonly notificationEmail: string;
+
+  /**
+   * The admin group associated with your Active Directory or IAM Identity Center account. This field is required as IAM_IDENTITY_CENTER is
+   * the only supported authentication method of the new Amazon QuickSight account
+   */
+  readonly adminGroup: string[];
+
+  /**
+   * The author group associated with your IAM Identity Center account.
+   */
+  readonly authorGroup: string[];
+
+  /**
+   * The reader group associated with your IAM Identity Center account.
+   */
+  readonly readerGroup: string[];
+
+  /**
+  * The region to use as main QuickSight region (used to store configuration and identities info)
+  */
+  readonly identityRegion: string;
+
+  private readonly removalPolicy: RemovalPolicy;
+
+  private readonly serviceToken: string;
+  private readonly policyActions: string[];
+
+  constructor (scope: Construct, id: string, props: QuickSightSubscriptionProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: QuickSightSubscription.name,
+    };
+    super(scope, id, trackedConstructProps);
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    this.accountName = props.accountName;
+    this.notificationEmail = props.notificationEmail;
+    this.adminGroup = props.adminGroup;
+    this.authorGroup = props.authorGroup;
+    this.readerGroup = props.readerGroup;
+    this.identityRegion = props.identityRegion;
+
+    this.policyActions = [
+      'quicksight:Subscribe',
+      'quicksight:UpdateAccountSettings',
+      'quicksight:Create*',
+      'quicksight:Unsubscribe',
+      'quicksight:DescribeAccountSubscription',
+      'sso:GetManagedApplicationInstance',
+      'sso:CreateManagedApplicationInstance',
+      'sso:GetManagedApplicationInstance',
+      'sso:DeleteManagedApplicationInstance',
+      'sso:GetManagedApplicationInstance',
+      'sso:DescribeGroup',
+      'sso:SearchGroups',
+      'sso:GetProfile',
+      'sso:AssociateProfile',
+      'sso:DisassociateProfile',
+      'sso:ListProfiles',
+      'sso:ListDirectoryAssociations',
+      'sso:DescribeRegisteredRegions',
+    ];
+
+    if (props.authenticationMethod != 'IAM_IDENTITY_CENTER') {
+      this.policyActions = this.policyActions.concat(
+        [
+          'ds:AuthorizeApplication',
+          'ds:UnauthorizeApplication',
+          'ds:CheckAlias',
+          'ds:CreateAlias',
+          'ds:DescribeDirectories',
+          'ds:DescribeTrusts',
+          'ds:DeleteDirectory',
+          'ds:CreateIdentityPoolDirectory',
+        ],
+      );
+    }
+
+    this.executionRole = new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaVPCAccessExecutionRole'),
```

Comment:
We avoid using managed roles as much as possible, can you check the MSK PR for how to build this policy?

Reply:
This is the [LambdaExecution role](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AWSLambdaVPCAccessExecutionRole.html) that as we clarified is already added by the CR Provider so will remove it, I am adding QS specific policies as an inline policy (see [lines 120-173 on quicksight-subscription.ts](https://github.com/awslabs/data-solutions-framework-on-aws/pull/505/commits/e5be8f31c3b1b9a7d3d5afa4b75baca3a3abd406#diff-01d4a3316c378311bf0ac590ebf03c3bddb557d8ab9fe5b5b36b9abbb2754904R120-R173) . Let me know if this is ok or shall we use customer managed policies for this.